numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -4060,6 +4060,22 @@ _vec_string(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *NPY_UNUSED(kw
     else if (PyArray_TYPE(char_array) == NPY_UNICODE) {
         method = PyObject_GetAttr((PyObject *)&PyUnicode_Type, method_name);
     }
+    else if (NPY_DT_is_user_defined(PyArray_DESCR(char_array))) {
+        PyArray_DTypeMeta* DType = NPY_DTYPE(PyArray_DESCR(char_array));
+        PyTypeObject* scalar_type = DType->scalar_type;
+        PyObject* allowed_scalar_types = PyTuple_Pack(2, (PyObject*)&PyBytes_Type, (PyObject*)&PyUnicode_Type);
+        int is_sub = PyObject_IsSubclass(
+            (PyObject*)scalar_type, allowed_scalar_types);
+        Py_DECREF(allowed_scalar_types);
+        if (is_sub == 1) {
+            method = PyObject_GetAttr((PyObject*)scalar_type, method_name);
+        } else {
```

Comment:
```suggestion
        } else
        {
```
Just to note, I don't care.

Reply:
Arrrg! sorry, I typed the above, it should have been:
```
}
else {
```
:facepalm: